crux-core/src/crux/kv/index_store.clj

```diff
@@ -867,25 +909,58 @@
 
 ;;;; IndexStore
 
+(defn- val-idxs [v]
+  (let [val->idxs (HashMap.)]
+    (cond
+      (vector? v) (dorun
+                   (map-indexed (fn [idx el]
+                                  (.compute val->idxs el
+                                            (reify BiFunction
+                                              (apply [_ _ acc]
+                                                (let [^List acc (or acc (ArrayList.))]
```

Comment:
Similar thing here, the actual elements here are ints, so there will be boxing. Worth testing indexing and pull of documents with large vectors inside to sanity check if it matters or not, but that applies more broadly to the entire approach as well.